File: Samples.Objc/Samples.ObjC/ViewControllers/CheckoutViewController/Views/AddItemCell.m
//
//  AddItemCell.m
//  PayPalNativeCheckoutObjC
//
//  Created by Nguyen, The Nhat Minh on 3/9/21.
//  Copyright © 2020 PayPal. All rights reserved.
//

#import "AddItemCell.h"
#import "UIViewController+Extension.h"

@interface AddItemCell ()

@property (nonatomic) UILabel *addItemLabel;

@end

@implementation AddItemCell

- (instancetype)init {
  self = [super init];
  self.selectionStyle = UITableViewCellAccessoryNone;
  [self setupUI];
  [self setupConstraints];
  return self;
}

- (void)setupUI {
  self.addItemLabel = [UIViewController labelWithText:@"Add item" font:[UIFont systemFontOfSize:16] color:[UIColor systemBlueColor]];
  self.addItemLabel.translatesAutoresizingMaskIntoConstraints = false;
  
  [self.contentView addSubview:self.addItemLabel];
}

- (void)setupConstraints {
  [[self.addItemLabel.centerXAnchor constraintEqualToAnchor:self.contentView.centerXAnchor] setActive:true];
  [[self.addItemLabel.centerYAnchor constraintEqualToAnchor:self.contentView.centerYAnchor] setActive:true];
}

@end

File: Samples.Swift/Samples.Swift/ViewControllers/CheckoutViewController/Views/AddItemCell.swift
//
//  AddItemCell.swift
//  paypalcheckout-iOS-swift-sample
//
//  Created by Jax DesMarais-Leder on 3/10/21.
//  Copyright © 2021 PayPal. All rights reserved.
//

import UIKit

class AddItemCell: UITableViewCell {

  lazy var addItemLabel: UILabel = {
    let label = UILabel()
    label.font = UIFont.boldSystemFont(ofSize: 16)
    label.textColor = .systemBlue
    label.text = "Add Item"
    label.sizeToFit()
    label.translatesAutoresizingMaskIntoConstraints = false
    return label
  }()
  
  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
    super.init(style: style, reuseIdentifier: reuseIdentifier)
    
    contentView.addSubview(addItemLabel)
    NSLayoutConstraint.activate(getConstraints())
  }
  
  required init?(coder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  func getConstraints() -> [NSLayoutConstraint] {
    [
      addItemLabel.centerXAnchor.constraint(equalTo: contentView.centerXAnchor),
      addItemLabel.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
    ]
  }
}
